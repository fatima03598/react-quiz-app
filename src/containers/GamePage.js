import React from 'react';
import { render } from 'react-dom';
import {Button} from 'reactstrap';
import Question from '../components/Question'

class GamePage extends React.Component {
    constructor(){
        super()
        this.state={
            playersNumber: null,
            difficulty: null,
            category: null,
            questionsArray: [],
            userNames: {
                
            },
            userScores: {

            },
            totalScore: 0
        }
    }
    


    addUserNames = (num) => {
        let html = []
        for(let i=0; i<num; i++) {
            html.push(<label>{`Player ${i+1}`}<input type="text" value={this.state.userNames[i]} name={`${i}`} onChange={this.updateUsers}/></label>)
        }
        
        return html
    }

    updateUsers = (e) => {
        // e.preventDefault();
        const obj = e.target.name
        const name = e.target.value
        this.setState({ userNames: { ...this.state.userNames, [obj]: name}})
        this.setState({ userScores: { ...this.state.userScores, [name]: 0}})
    }


    componentDidMount() {

            this.setState({
                playersNumber: this.props.match.params.playersNumber,
                difficulty: this.props.match.params.difficulty, 
                category: this.props.match.params.category
            });
       
       //call api within componenetDidMount();
    }

    getQuestions = async (e) => {
        e.preventDefault();
        if(!this.state.category){
            console.log('error in getQuestions, category does not exist')
        }
        const url = `https://opentdb.com/api.php?amount=10&category=${this.state.category}&difficulty=${this.state.difficulty}&type=multiple&encode=base64`;
        console.log(url);
        const response = await fetch(url);
        const data = await response.json();
        this.setState({questionsArray: data.results});
        console.log(data);
        

    }

    totalScore = (n, user) => {
        this.setState({userScores : {...this.state.userScores, [user]: this.state.userScores[user] + n}})
    }

    render()  {
        console.log(this.state.questionsArray);
        console.log(this.state.userNames);
        return(
            <div className='GamePage'>
                <h1>Game Page</h1>
                <h2>Total Score: {this.state.totalScore} </h2>
                <div>
                    <h2>Scores:</h2>
                    {(Object.values(this.state.userNames)).map(user => {
                        return (<h3>{user}:{this.state.userScores[user]}</h3>)
                    })}
                </div>
                <form>
                    {this.state.questionsArray.map((question, i) => <Question questionContent = {question} key={i} id={i} totalScore = {this.totalScore} playersNumber={this.state.playersNumber} userNames={this.state.userNames}/>)}
                </form>
                <form className="name-form">
                    <h3>Enter player name{this.state.playersNumber > 1? "s":""}</h3>
                    {this.addUserNames(this.state.playersNumber)}
                    <input type="submit" onClick={this.getQuestions} value="Start Game" />

                </form>
            </div>
        );
    }
   
}

export default GamePage;