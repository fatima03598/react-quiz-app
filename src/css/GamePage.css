.game-page-container{
    display: flex;
    flex-direction: column;
    align-items: center;
    background-color: #90CAFD;
    height: 100%;
}

.name-form{
    /* height: 100%; */
    padding: 5em;
    /* height: 30em; */
    width: 20em;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    align-items: center;
}

.name-input {
    width: 100%;
    display: flex;
    font-size: 1em;
    border-radius: 1em;
}
.name-input-box {
    margin: 1em;
    border-radius: 1em;
    padding: 0.5em;
    border: 0;
}

.start-game-button-2 {
    border-radius: 1em;
    border: 0;
    padding: 1em;
    background-color: #004480;
    color: white;
}
.name-form-title{
    font-size: 2em;
    margin: 1em 0;
}