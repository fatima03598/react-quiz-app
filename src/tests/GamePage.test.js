import React from 'react';
import { BrowserRouter, Link } from 'react-router-dom';
import { shallow, mount } from 'enzyme';
import GamePage from '../containers/GamePage';
jest.mock('./__mock__/Request');
import Request from './__mock__/Request';
import ApiObject from './__mock__/__mockData__.json';

describe('GamePage', () => {

    // let wrapper;
    // beforeEach(() => {
     
    // });

    it('GamePage should render GamePage div', () => {
        const wrapper = shallow(<GamePage  required={true}
    match={{params: {playersNumber: 1}, category: 'General Knowledge', difficulty:'medium'}}/>); 
        expect(wrapper.exists('.GamePage')).toEqual(true);
    });


    it('GamePage should load user data', () => {
        return GamePage.getQuestions()
        .then(data => {
          expect(data).toBeDefined()
          expect(data.entity.name).toEqual('Koen van Gilst')
        })
    });

    it('GamePage should load user data', () => {
      
          expect(Request).toEqual(ApiObject);
    });




  })