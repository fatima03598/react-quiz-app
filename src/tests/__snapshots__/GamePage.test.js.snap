// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GamePage /> matches the snapshot 1`] = `
<div
  className="game-page-container"
>
  <form
    className="name-form"
  >
    <div
      className="name-form-title"
    >
      <label>
        Enter player name
        
      </label>
    </div>
    <div
      className="name-input"
    >
      <label>
        Player 1:
        <input
          className="name-input-box"
          name="0"
          onChange={[Function]}
          placeholder="Name"
          required={true}
          type="text"
        />
      </label>
    </div>
    
  </form>
</div>
`;
